xbmc/cores/VideoPlayer/VideoPlayer.cpp

```diff
@@ -2821,6 +2821,15 @@ void CVideoPlayer::HandleMessages()
         std::shared_ptr<CInputStreamPVRBase> pvrinputstream = std::static_pointer_cast<CInputStreamPVRBase>(m_pInputStream);
         pvrinputstream->Pause(speed == 0);
       }
+      
+      if (speed != m_playSpeed &&
+          (speed == DVD_PLAYSPEED_PAUSE || m_playSpeed == DVD_PLAYSPEED_PAUSE))
+      {
+        double iTime = m_VideoPlayerVideo->GetCurrentPts();
+        if (iTime == DVD_NOPTS_VALUE)
+          iTime = m_clock.GetClock();
+        m_pInputStream->Pause(iTime);
```

Comment:
With here to take the `Pause(time)` seems not work, if a addon not have a own demux.

As this also here https://github.com/xbmc/xbmc/blob/master/xbmc/cores/VideoPlayer/DVDDemuxers/DVDDemuxFFmpeg.cpp#L718 becomes it two times called then and the paused stream think "Pause" is ended.

Reply:
Cool, will add demuxer to check.

Reply:
iTime also needs to divided by 1000.